atomic_reactor/util.py

```diff
@@ -985,8 +985,93 @@ def get_manifest_list(image, registry, insecure=False, dockercfg_path=None):
     return response
 
 
+def get_all_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
+                             versions=('v1', 'v2', 'v2_list', 'oci', 'oci_index')):
+    """Return manifest digests for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param registry: str, URI for registry, if URI schema is not provided,
+                          https:// will be used
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+
+    :return: dict of successful responses, with versions as keys
+    """
+    digests = {}
+    registry_session = RegistrySession(registry, insecure=insecure, dockercfg_path=dockercfg_path)
+    for version in versions:
+        response, _ = get_manifest(image, registry_session, version)
+        if response:
+            digests[version] = response
+
+    return digests
+
+
+def get_inspect_for_image(image, registry, insecure=False, dockercfg_path=None):
+    """Return inspect for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param registry: str, URI for registry, if URI schema is not provided,
+                          https:// will be used
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+
+    :return: dict of inspected image
+    """
+    all_man_digests = get_all_manifest_digests(image, registry, insecure=insecure,
+                                               dockercfg_path=dockercfg_path)
```

Comment:
Maybe restrict versions here to those we can handle?

Reply:
yeah will restring it in that function